lib/discover/discover.go

```diff
@@ -303,8 +307,8 @@ func (d *Discoverer) Announcement() Announce {
 
 func (d *Discoverer) announcementPkt(allowExternal bool) Announce {
 	var addrs []string
-	if d.extPort != 0 && allowExternal {
-		addrs = []string{fmt.Sprintf("tcp://:%d", d.extPort)}
+	if allowExternal && d.extAddr != nil {
```

Comment:
Ah, sorry. Yeah, that would have the same issue, but in practice it doesn't as it's never nil in syncthing (we always have address lister thingy), the only reason I even do the check is that in some tests we do instantiate it with a literal nil there (and that works to).